utility/firebase.js

```diff
@@ -310,3 +311,55 @@ export const subscribeToCMSStatus = (dateCallback) => {
       dateCallback(offUntilDate);
     });
 };
+
+export const getActiveHackathon = db
+  .collection(InternalWebsitesCollection)
+  .doc(LivesiteCollection)
+  .get()
+  .then((doc) => doc.data()?.activeHackathon);
+
+export const subscribeToLivesiteAnnouncements = (hackathon, callback) => {
+  return db
+    .collection(Hackathons)
+    .doc(hackathon)
+    .collection('Announcements')
+    .orderBy('timestamp', 'desc')
+    .onSnapshot((querySnapshot) => {
+      const announcements = {};
+      querySnapshot.docs.forEach((doc) => {
+        announcements[doc.id] = doc.data();
+      });
+      callback(announcements);
+    });
+};
+
+export const updateAnnouncement = async (hackathon, announcement) => {
+  if (announcement.id) {
+    const ref = db
+      .collection(Hackathons)
+      .doc(hackathon)
+      .collection('Announcements')
+      .doc(announcement.id);
+    delete announcement.id;
+    await ref.set(announcement);
+    return announcement.id;
+  }
+  announcement.timestamp = Date.now();
+  const ref = await db
+    .collection(Hackathons)
+    .doc(hackathon)
+    .collection('Announcements')
+    .doc()
+    .set(announcement);
+  console.log(ref);
+  return ref;
+};
+
+export const deleteAnnouncement = async (hackathon, id) => {
+  await db
+    .collection(Hackathons)
+    .doc(hackathon)
+    .collection('Announcements')
```

Comment:
is there a way to refactor this to pull out `db.collection(Hackathons).doc(hackathon).collection('Announcements')` into its own function or constant?

Reply:
something similar to `getActiveHackathon`

Reply:
G o o d c o d e